synchronizer/synchronizer.go

```diff
@@ -1309,7 +1320,17 @@ func (s *ClientSynchronizer) processTrustedBatch(trustedBatch *types.Batch, dbTx
 				if isBatchClosed {
 					//Sanity check
 					if s.trustedState.lastStateRoot != nil && trustedBatch.StateRoot != *s.trustedState.lastStateRoot {
-						s.halt(s.ctx, fmt.Errorf("stateRoot calculated (%s) is different from the stateRoot (%s) received during the trustedState synchronization", *s.trustedState.lastStateRoot, trustedBatch.StateRoot))
+						if trustedBatchL2Data.Hex() != "0x"+common.Bytes2Hex(batches[0].BatchL2Data) {
```

Comment:
I think is better to log always trustedBatchL2Data, batches[0].BatchL2Data and contents of storedTxs and syncedTxs, not only to log it if trustedBatchL2Data != batches[0].BatchL2Data, in this way we will have all the log info whatever is the scenario